supportedTransports.js

```diff
@@ -0,0 +1,96 @@
+const { default: LedgerTransportU2F } = require('@ledgerhq/hw-transport-u2f');
+const { default: LedgerTransportWebUsb } = require('@ledgerhq/hw-transport-webusb');
+const { default: LedgerTransportWebHid } = require('@ledgerhq/hw-transport-webhid');
+const platform = require('platform');
+
+let ENABLE_DEBUG_LOGGING = false;
+const debugLog = (...args) => {
+    ENABLE_DEBUG_LOGGING && console.log(...args)
+};
+
+// Fallback order inspired by https://github.com/vacuumlabs/adalite
+async function isWebUsbSupported() {
+    try {
+        const isSupported = await LedgerTransportWebUsb.isSupported();
+        return isSupported && platform.os.family !== 'Windows' && platform.name !== 'Opera';
+    } catch (e) {
+        return false;
+    }
+}
+
+async function isWebHidSupported() {
+    return LedgerTransportWebHid.isSupported()
+        .catch(() => false);
+}
+
+async function isU2fSupported() {
+    return LedgerTransportWebHid.isSupported()
+        .catch(() => false);
+}
+
+async function createSupportedTransport() {
+    const [
+        supportWebHid,
+        supportWebUsb,
+        supportU2f
+    ] = await Promise.all([
+        isWebHidSupported(),
+        isWebUsbSupported(),
+        isU2fSupported()
+    ]);
+
+    debugLog("Transports supported:", { supportWebHid, supportWebUsb, supportU2f });
+
+    if (!supportWebHid && !supportWebUsb && !supportU2f) {
+        const err = new Error('No transports appear to be supported.');
+        err.name = 'NoTransportSupported';
+        throw err;
+    }
+
+    // Sometimes transports return true for `isSupported()`, but are proven broken when attempting to `create()` them.
+    // We will try each transport we think is supported in the current environment, in order of this array
+    const supportedTransports = [
+        // WebHID is still supported by latest Chrome, so try that first
+        ...(supportWebHid && [{ name: 'WebHID', createTransport: () => LedgerTransportWebHid.create() }]),
+
+        ...(supportWebUsb && [{ name: 'WebUSB', createTransport: () => LedgerTransportWebUsb.create() }]),
+
+        // Firefox/Mozilla intend to not support WebHID or WebUSB
+        ...(supportU2f && [{ name: 'U2F', createTransport: () => LedgerTransportU2F.create() }]),
```

Comment:
For any non-supported transport, this resolves to `...false` which is invalid logic.

Should be:

```suggestion
        ...(supportWebHid ? [{ name: 'WebHID', createTransport: () => LedgerTransportWebHid.create() }] : []),

        ...(supportWebUsb ? [{ name: 'WebUSB', createTransport: () => LedgerTransportWebUsb.create() }] : []),

        // Firefox/Mozilla intend to not support WebHID or WebUSB
        ...(supportU2f ? [{ name: 'U2F', createTransport: () => LedgerTransportU2F.create() }] : []),
```